i18n/checkLocaleKeys.js

```diff
@@ -0,0 +1,73 @@
+import fs from 'fs'
+import path from 'path'
+import { fileURLToPath } from 'url'
+
+// Convert import.meta.url to __dirname equivalent
+const __filename = fileURLToPath(import.meta.url)
+const __dirname = path.dirname(__filename)
+
+// Define the path to our i18n locales directory
+const localesI18nDirectory = path.join(
+    __dirname,
+    '/locales'
+)
+
+// Read all JSON files from the locales directory
+let localeFiles = fs
+    .readdirSync(localesI18nDirectory)
+    .filter(file => file.endsWith('.json'))
+
+// Ensure en.json is the first file
+const enFile = 'en.json'
+if (!localeFiles.includes(enFile)) {
+    console.error('en.json file not found in the locales directory')
+    process.exit(1)
+}
+localeFiles = localeFiles.filter(file => file !== enFile)
+localeFiles.unshift(enFile)
+
+// This function extracts the keys from the JSON object
+const getKeys = (obj, prefix = '') =>
+    Object.keys(obj).reduce((keys, key) => {
+        const fullKey = prefix ? `${prefix}.${key}` : key
+        if (typeof obj[key] === 'object' && obj[key] !== null) {
+            keys.push(...getKeys(obj[key], fullKey))
+        } else {
+            keys.push(fullKey)
+        }
+        return keys
+    }, [])
+
+// This reads and parses the JSON files
+const localeKeys = localeFiles.reduce((acc, file) => {
```

Comment:
optimization time! since we're iterating over the localfiles + the keys in each file, we have O(n+m)^2 (I think). Instead, we can make this just O(n+m).
Basically, you're looping over the localFiles + the keys multiple times. Instead, just loop over the localeFiles and keys in one big reduce function. When you pull each key, check it, then add it to the `missingKeys` object if it's missing. 

```
.reduce((missingKeys, file) => { 
...
missingKeys[fileName].push(key)
...
  })
```